source/server/listener_impl.cc

```diff
@@ -551,8 +557,11 @@ void ListenerImpl::buildUdpListenerFactory(Network::Socket::Type socket_type,
     // looking at the GSO code there are substantial copying inefficiency so I don't think it's
     // wise to enable to globally for now. I will circle back and fix both of the above with
     // a non-QUICHE GSO implementation.
-    if (Api::OsSysCallsSingleton::get().supportsUdpGso()) {
-      udp_listener_config_->writer_factory_ = std::make_unique<Quic::UdpGsoBatchWriterFactory>();
+    if (!udp_listener_config_->writer_factory_.has_value() &&
+        Api::OsSysCallsSingleton::get().supportsUdpGso()) {
+      udp_listener_config_->writer_factory_ =
+          *Registry::FactoryRegistry<Network::UdpPacketWriterFactory>::getFactoryByType(
```

Comment:
For here any below, the writer factory is not config dependent, so might just instantiate it directly as before?

Reply:
Oh, that's right. Hmm, it's also a bit weird to hold a handle to a registered factory. 
In most use cases, there are two layer of factories. The registered one only has a default constructor, while the other one can have non-default constructors which take config knobs, so the relevant configs are retained in the secondary factory. i.e. QuicServerTransportSocketConfigFactory and QuicServerTransportSocketFactory. I thought this would be an over kill for writer because right now both configs for GSO and raw UDP writers are just empty proto. But the weirdness of the code here makes me wonder if it worth to have the extra layer.

Reply:
What would be the benefit of adding an extra layer here?

Reply:
writer_factory_ can be owned by udp_listener_config_. And it will allow any future extension to take configuration.

Reply:
Thanks for the helpful offline discussion! I think I've done the needful and we now have a FactoryFactory.

Reply:
@danzh2010 friendly ping

Reply:
Nice! It actually reduced the lines need to be changed here.